client-spark/spark-3/src/main/java/org/apache/spark/shuffle/celeborn/SortBasedShuffleWriter.java

```diff
@@ -70,7 +70,8 @@
   private final int numPartitions;
 
   private final long pushBufferMaxSize;
-  private SortBasedPusher sortBasedPusher;
+  private SortBasedPusher[] pushers = new SortBasedPusher[2];
```

Comment:
Two data pushers might be not enough, maybe adding a configuration will be better.

Reply:
Double buffering typically uses two buffers